lib/clouds/shared_functions.py

```diff
@@ -1247,38 +1079,115 @@ def populate_cloudconfig(self, obj_attr_list) :
                         cloudconfig += "      up /etc/openvpn/client_connected.sh\n"
                 fh.close()
 
-        cloudconfig += """
-runcmd:
-  - chmod +x /tmp/userscript.sh"""
-
-        # We can't run the userdata from cloudbench until the VPN is connected,
-        # so only run it if we're not using the VPN.
-        # Otherwise, /etc/openvpn/client_connected.sh will do it.
-        if obj_attr_list["use_vpn_ip"].lower() == "false" :
-            cloudconfig += """
-  - /tmp/userscript.sh"""
-        else :
-            cloudconfig += """
-  - chmod +x /etc/openvpn/client_connected.sh
-  - mv """ + conf_destination + """ /tmp/cbvpn.conf
-  - rm -f /etc/openvpn/*.conf /etc/openvpn/*.ovpn
-  - mv /tmp/cbvpn.conf """ + conf_destination + """
-  - mkdir -p /var/log/openvpn
-  - openvpn --daemon --config """ + conf_destination + """
-"""
-  #- systemctl start openvpn@""" + obj_attr_list["cloud_name"] + """_client-cb-openvpn-cloud.service
-  #- service openvpn start
+        if obj_attr_list["userdata_post_boot"].lower() == "true" or obj_attr_list["use_vpn_ip"].lower() != "false" :
+            cloudconfig += "\nruncmd:\n"
+            cloudconfig += "  - chmod +x /tmp/cb_post_boot.sh\n"
```

Comment:
Why is this cb_post_boot? That is not the same as the real post_boot is it? Userdata size has a distinct limit on Amazon and making the userdata too large will cause the userdata upload to fail.

Reply:
So, one of the possible options (on a per-cloud basis) is to run the actual `cb_post_boot.sh` via cloud-init. It does not increase the size of the userdata, since all the actual data is obtained via an rsync (also on the cloud-init).

Reply:
I think you misunderstood. I was simply asking why you renamed it to post_boot......... because that is confusing with the original post_boot. I definitely don't want to run the original post boot over cloud-init. Let's keep it as is. Do you have a strong reason to rename the userscript to "post_boot"? Can you bring back the original name so that it does not clash with our current post boot script?